lib/Transforms/Loops/SeaSCEVUtils.cc

```diff
@@ -0,0 +1,41 @@
+#include "llvm_seahorn/Loops/SeaSCEVUtils.h"
+
+#include "llvm/Analysis/ScalarEvolutionAliasAnalysis.h"
+
+namespace llvm_seahorn {
+
+bool seaSCEVContainsMul(const llvm::SCEV *Expr) {
+  using namespace llvm;
+  assert(Expr);
+
+  if (const auto *M = dyn_cast<SCEVMulExpr>(Expr)) {
+    // If not all multiplication operands are constants we consider
```

Comment:
I think this checks that at most one operand is non-constant. This is what I would expect.
But the comment seems to indicate that we only allow the case where all operands are are constants.